diff --git a/configure.ac b/configure.ac
index 282cf4618f..c847a43a99 100644
--- a/configure.ac
+++ b/configure.ac
@@ -568,26 +568,6 @@ AS_IF([test "$GCC" = yes], [

     : ${MJIT_HEADER_FLAGS='-P -dD'}

-    # -fstack-protector
-    AS_CASE(["$target_os"],
-    [mingw*], [
-	stack_protector=no
-    ])
-    AS_IF([test -z "${stack_protector+set}"], [
-	AS_FOR(option, opt, [-fstack-protector-strong -fstack-protector], [
-	    RUBY_TRY_CFLAGS(option, [stack_protector=yes])
-	    AS_IF([test "x$stack_protector" = xyes], [
-		RUBY_TRY_LDFLAGS(option, [], [stack_protector=])
-	    ])
-	    AS_IF([test "x$stack_protector" = xyes], [stack_protector=option; break])
-	])
-    ])
-    AS_CASE(["$stack_protector"], [-*], [
-	RUBY_APPEND_OPTION(XCFLAGS, $stack_protector)
-	RUBY_APPEND_OPTION(XLDFLAGS, $stack_protector)
-	RUBY_APPEND_OPTION(LDFLAGS, $stack_protector)
-    ])
-
     AS_CASE("${compress_debug_sections:-zlib}",
     [none|no], [], [
     RUBY_TRY_LDFLAGS(${linker_flag}--compress-debug-sections=${compress_debug_sections:-zlib},
